pkg/manager/diff.go

```diff
@@ -224,33 +244,64 @@ loop:
 	return g.Wait()
 }
 
-func (dc *diffContext) waitCorpusTriage(ctx context.Context) {
-	// Wait either until we have triaged 90% of the candidates or
-	// once MaxTriageTime has passed.
-	// We don't want to wait for 100% of the candidates because there's usually a long
-	// tail of slow triage jobs + the value of the candidates diminishes over time.
-	const triagedThreshold = 0.9
+func (dc *diffContext) waitCorpusTriage(ctx context.Context, threshold float64) chan struct{} {
 	const backOffTime = 30 * time.Second
-
-	startedAt := time.Now()
-	for {
-		select {
-		case <-time.After(backOffTime):
-		case <-ctx.Done():
-			return
-		}
-		triaged := dc.new.triageProgress()
-		if triaged >= triagedThreshold {
-			log.Logf(0, "triaged %.1f%% of the corpus", triaged*100.0)
-			return
-		}
-		if dc.cfg.MaxTriageTime != 0 &&
-			time.Since(startedAt) >= dc.cfg.MaxTriageTime {
-			log.Logf(0, "timed out waiting for %.1f%% triage (have %.1f%%)",
-				triagedThreshold*100.0, triaged*100.0)
-			return
+	ret := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-time.After(backOffTime):
+			case <-ctx.Done():
+				return
+			}
+			triaged := dc.new.triageProgress()
+			if triaged >= threshold {
+				log.Logf(0, "triaged %.1f%% of the corpus", triaged*100.0)
+				close(ret)
+				return
+			}
 		}
+	}()
+	return ret
+}
+
+var ErrPatchedAreaNotReached = errors.New("fuzzer has not reached the patched area")
+
+func (dc *diffContext) monitorPatchedCoverage(ctx context.Context) error {
+	if dc.cfg.FuzzToReachPatched == 0 {
+		// The feature is disabled.
+		return nil
+	}
+	if len(dc.new.cfg.Experimental.FocusAreas) < 2 {
```

Comment:
TODO: check how difficult it is to check if the coverage filter actually matched any PCs.

Reply:
Added one more commit that does it.

Reply:
I don't see any tests here. If you want to move faster w/o tests, please add some TODOs or mention what tests are needed in some issue.